src/wallet/wallet.cpp

```diff
@@ -1495,11 +1497,17 @@ bool CWallet::IsWalletFlagSet(uint64_t flag)
     return (m_wallet_flags & flag);
 }
 
-void CWallet::SetWalletFlags(uint64_t overwriteFlags, bool memonly)
+bool CWallet::SetWalletFlags(uint64_t overwriteFlags, bool memonly)
 {
     m_wallet_flags = overwriteFlags;
+    if ( (overwriteFlags >> 32 | g_known_wallet_flags >> 32) != (g_known_wallet_flags >> 32) ) {
+        // contains unkown non-tolerable wallet flags
+        return false;
```

Comment:
I wonder if this should `throw` here instead of being a `can-fail` method.

Reply:
I think a specific error handling makes more sense (see error handling in walletdb)